lib/improver/wind_direction.py

```diff
@@ -0,0 +1,237 @@
+
+        return dist_from_mean_norm
+
+    @staticmethod
+    def wind_dir_decider(wind_dir_deg, wind_dir_deg_mean, r_vals):
+        """If the wind direction is so widely scattered that the r value
+           is nearly zero then this indicates that the average wind direction
+           is essentially meaningless.
+           We therefore subistuite this meaningless average wind
+           direction value for the wind direction taken from the first
+           ensemble member."""
+
+        # Threshold r value.
+        r_thresh = 0.01
+
+        # Mask True if r values below threshold.
+        where_low_r = np.where(r_vals < r_thresh, True, False)
+
+        # If the whole array contains good r-values, return origonal array.
+        if not where_low_r.any():
+            return wind_dir_deg_mean
+
+        # Takes first ensemble member.
+        first_member = wind_dir_deg[0]
+
+        # If the r-value is low - subistite average wind direction value for
+        # the wind direction taken from the first ensemble member.
+        reprocessed_wind_dir_mean = np.where(where_low_r, first_member,
+                                             wind_dir_deg_mean)
+
+        return reprocessed_wind_dir_mean
+
+    def process(self, cube_ens_wdir):
+        """
+        Create a cube containing the wind direction averaged over the ensemble
+        members.
+
+        Args:
+            cube_wdir (iris.cube.Cube):
+                Cube containing wind direction from multiple ensemble members.
+
+        Returns:
+            result (iris.cube.Cube):
+                Cube containing the wind direction averaged from the
+                ensemble members.
+
+        Raises
+        ------
+        TypeError: If cube_wdir is not a cube.
+
+        """
+
+        if not isinstance(cube_ens_wdir, iris.cube.Cube):
+            msg = "Wind direction input is not a cube, but {}"
+            raise TypeError(msg.format(type(cube_ens_wdir)))
+
+        # Force input cube to float32.
+        enforce_float32_precision(cube_ens_wdir)
+
+        # Creates result cube - copies input and removes realization dimension.
+        for cube_mean_wdir in cube_ens_wdir.slices_over("realization"):
+            cube_mean_wdir.remove_coord('realization')
+            break
+
+        y_coord_name = cube_ens_wdir.coord(axis="y").name()
+        x_coord_name = cube_ens_wdir.coord(axis="x").name()
+        wind_dir_deg = next(cube_ens_wdir.slices(["realization",
+                                                  y_coord_name,
+                                                  x_coord_name])).data
+
+        # Convert wind direction from degrees to complex numbers.
+        wind_dir_complex = WindDirection.deg_to_complex(wind_dir_deg)
+
+        # Find the complex average - which actually signifies a point between
+        # all of the data points in POLAR co-cordinates (not degrees).
+        wind_dir_complex_mean = np.mean(wind_dir_complex, axis=0)
```

Comment:
This axis = 0 is a bit dangerous, you're assuming we want to find the mean along the first dimension. What you could do is pass in the name of the coordinate you want to find the mean of and then use the iris function `cube.coord_dims(coord)` to find the axis that coordinate is associated with. It's a bit more complex but safer.